### DISCUSSION TITLE
UBL (15x15) Meshing w/ Probe Settings

### DISCUSSION THREAD

**Discussion by hkasap83** [Jyers/Marlin#417] [Q&A] [answered]
First of all, thank you for this great job and to all friends who shared their experience to solve our problems. I got a BL Touch probe a few days ago. Im trying to use UBL 15x15 meshing (4.2.2 main board) with my BL touch. 

My first question, in the UBL settings menu there is a “convert mesh to plane” item. Is this the last step have to do for using the probed mesh or is it the only visual thing that does not affect anything?

Second question, i did the manuel bed level for four corners as i always do and my bed center was lower than the four corners as always. Then, i set the x,y,z-offset for the probe installation but when i did this, four corners were affected as expected because of the z-offset for the z-home (four corners became more close to the nozle). So, when u r using the probe how should be the initial bed level? Are u doing any manuel bed level before the ubl meshing?

**Comment by SikVac** [comment]
I have the same questions.
Will we get an answer?

**Comment by smooreace** [comment]
You should always start with the bed TRAMMED as best as possible manually
prior to doing any form of automated bed scanning

**Reply by gothcha** [reply]
You do not need to use the 'convert mesh to plane' for the bed levelling to operate correctly. Just ensure that after the mesh is complete you save the mesh to EEPROM from the levelling menu so it will be available on next power-up.

**Reply by hkasap83** [reply]
ok, but what are we going to use it for?

**Reply by gothcha** [reply]
I am also not clear on what the intended use for the convert mesh to plane is, from the code it appears to generate a least squares fit to the mesh and create a flat plane, then enable with an M420 S1. Maybe @Jyers can elaborate?

**Reply by Jyers** [reply]
The author of the PR for the UBL systems added that. I debated removing it, but decided to leave it in case anyone ever found a use for it. You're right, it just creates a plane using the mesh data.

**Comment by Jyers** [comment, marked as answer]
One last note, at the bottom of the manual leveling menu there is a z position option. This can be used to account for the width of your probe (ie paper for most people) when manually leveling or tramming the bed. If you notice all of the points are too close, simply increase the z position slightly until it matches your desired amount of friction when testing the points, and then proceed to level each of them as you would.
